web-api/src/persistence/postgres/cases/getCaseByDocketNumber.ts

```diff
@@ -1,19 +1,110 @@
+import { getCaseCorrespondenceByDocketNumber } from '@web-api/persistence/postgres/caseCorrespondences/getCaseCorrespondenceByDocketNumber';
+import { getCaseStatistics } from '@web-api/persistence/postgres/cases/statistics/getCaseStatistics';
+import { getCaseStatusHistory } from '@web-api/persistence/postgres/cases/getCaseStatusHistory';
+import { getCaseMetadataByDocketNumber } from '@web-api/persistence/postgres/cases/getCaseMetadataByDocketNumber';
 
-// Named getCaseByDocketNumberPostgres until the Dynamo getCaseByDocketNumber is removed
-export const getCaseByDocketNumberPostgres = async (docketNumber: string) => {
-  return await getDbReader(reader => {
-    return reader
-      .selectFrom('dwCase as c')
-      .leftJoin('dwWorkItem as d', 'd.docketNumber', 'c.docketNumber')
-      .leftJoin(
-        'dwCaseCorrespondence as co',
-        'c.docketNumber',
-        'co.docketNumber',
-      )
-      .where('c.docketNumber', '=', docketNumber)
-      .selectAll()
-      .select('c.docketNumber')
-      .execute();
+export const getCaseByDocketNumber = async ({
+  applicationContext,
+  docketNumber,
+  includeConsolidatedCases = true,
+  user = undefined, // Only needed to check permissions on sealed addresses for consolidated cases
+}: {
+  docketNumber: string;
+  applicationContext: ServerApplicationContext;
+  includeConsolidatedCases?: boolean;
+  user?: UnknownAuthUser;
+}): Promise<RawCase> => {
+  // These case items are no longer in dynamoDB
+  const SK_FILTER_OUT = ['work-item', 'correspondence', 'case'];
+
+  const dbCaseMetadata = await getCaseMetadataByDocketNumber({
+    docketNumber,
+  });
+  if (!dbCaseMetadata) {
+    throw new NotFoundError(`Case ${docketNumber} not found`);
+  }
+
+  const [
+    caseStatusHistory,
+    caseCorrespondences,
+    statisticsWithPenalties,
+    workItems,
+  ] = await Promise.all([
+    getCaseStatusHistory({ docketNumber }),
+    getCaseCorrespondenceByDocketNumber({
+      docketNumber,
+    }),
+    getCaseStatistics({ docketNumber }),
+    getWorkItemsByDocketNumber({
+      docketNumber,
+    }),
+  ]);
+
+  const caseItems = await queryFull({
+    ExpressionAttributeNames: {
+      '#pk': 'pk',
+    },
+    ExpressionAttributeValues: {
+      ':pk': `case|${docketNumber}`,
+    },
+    KeyConditionExpression: '#pk = :pk',
+    applicationContext,
+  }).then(items =>
+    items.filter(
+      item => !SK_FILTER_OUT.some(prefix => item.sk.startsWith(prefix)),
+    ),
+  );
+
+  let consolidatedCases: Omit<
+    RawCase,
+    'consolidatedCases' | 'correspondence' | 'docketEntries'
+  >[] = [];
+  if (includeConsolidatedCases) {
+    consolidatedCases = await getCasesMetadataWithCounselByLeadDocketNumber({
+      applicationContext,
+      leadDocketNumber: dbCaseMetadata.leadDocketNumber!,
+    });
+    if (user) {
+      consolidatedCases = consolidatedCases.map(c =>
+        formatSealedAddresses(c, user),
```

Comment:
Should this be elsewhere? Probably. Was it already here before, but it's hard to tell because of the git diff? Yes.